test/overlapped-checker/main.c

```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "../../deps/uv/include/uv.h"
+#include "../../deps/uv/src/uv-common.h"
+
+int main() {
+  uv_loop_t *loop = uv_default_loop();
+  uv_pipe_t pipe;
+
+  for (int i = 0; i <= 2; i++) {
+    if (uv_pipe_init(loop, &pipe, 0) || uv_pipe_open(&pipe, i)) {
+      abort();
+    }
+
+    int non_overlapped_flag = !!(pipe.flags & UV_HANDLE_NON_OVERLAPPED_PIPE);
```

Comment:
This is unreliable. Libuv's UNIX pipe code could add a flag with the same value as `UV_HANDLE_NON_OVERLAPPED_PIPE` (which is Windows-only) and then this code starts printing different numbers.